### ISSUE TITLE
PowerConnect Prompt Issue

### ISSUE THREAD

**Issue by etern4ever** [ytti/oxidized#228]
We have a couple of PowerConnect blade server switches where the prompt isn't being detected.

Log message

 raised Oxidized::PromptUndetect with msg "#015#012switchmodule> not matching configured prompt (?-mix:^([\w\s.@-]+[#>]\s?)$)"

System Description: Dell Ethernet Switch
System Model ID: PCM6348
Machine Type: PowerConnect M6348

**Comment by ytti** [comment]
Curious, I guess you expect your prompt to be
"switchmodule>"

But it looks like it saw "#15#012switchmodule>", you might want to try to add '#' in the bracket of the prompt, before '-'.

**Comment by etern4ever** [comment]
#15#012 isn't actually a part of the prompt though. I will try adjusting the prompt to include that.

**Comment by ytti** [comment]
You could try to set on debugging to store terminal session. Probably it pukes some trash sometimes and sometimes does not, unfortunately not too uncommon for the CLI implementation to be be crappy. 

Funnily enough 015 is \r and 012 is \n, not sure if that is coincidence.

**Comment by etern4ever** [comment]
I don't think it is a coincidence.  I tried adding # in there.  but that didn't fix anything.  It has to be a more implementation and they are sending newline characters odd.  

raised Oxidized::PromptUndetect with msg "#015#012switchmodule> not matching configured prompt (?-mix:^([\w\s.@#-]+[#>]\s?)$)"

**Comment by ytti** [comment]
you could try adding \r? in the beginning of the regexp prompt

On 25 October 2015 at 00:11, etern4ever <notifications@github.com> wrote:

> I don't think it is a coincidence. I tried adding # in there. but that
> didn't fix anything. It has to be a more implementation and they are
> sending newline characters odd.
>
> raised Oxidized::PromptUndetect with msg "#15
> <https://github.com/ytti/oxidized/pull/15>#012switchmodule> not matching
> configured prompt (?-mix:^([\w\s.@#-]+[#>]\s?)$)"
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ytti/oxidized/issues/228#issuecomment-150851764>.
>



-- 
  ++ytti

**Comment by etern4ever** [comment]
Tried that.. it didn't work.   Still digging into it.  I am sure it has to do with newline characters.  Just need a way to clean up the output from the device.

**Comment by etern4ever** [comment]
I have tried a lot of things.  I still can't get it to detect prompt.  Even if I change prompt to \S+.  Very odd.

The crash log shows a CR LF on one line, then the prompt on the next.
![2015-10-27 08_20_01-photos](https://cloud.githubusercontent.com/assets/7612193/10759135/a0451e9c-7c83-11e5-867b-9e5651f016e2.png)

**Comment by ytti** [comment]
Do you have input debug file, with session stores, I'd like to take a peek.

On 27 October 2015 at 15:21, etern4ever <notifications@github.com> wrote:

> I have tried a lot of things. I still can't get it to detect prompt. Even
> if I change prompt to \S+. Very odd.
>
> The crash log shows a CR LF on one line, then the prompt on the next.
> [image: 2015-10-27 08_20_01-photos]
> <https://cloud.githubusercontent.com/assets/7612193/10759135/a0451e9c-7c83-11e5-867b-9e5651f016e2.png>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ytti/oxidized/issues/228#issuecomment-151492154>.
>



-- 
  ++ytti

**Comment by etern4ever** [comment]
How do I enable this?   Under input section in the config, I have debug: /tmp/oxidized_log but this doesn't appear to create the logs files as I could expect.

**Comment by ytti** [comment]
I think we may have changed it so that it creates logs always in your oxidized dir with name of the device, so it's really 'true' or 'false' kinda deal now.

**Comment by etern4ever** [comment]
I have the crash files (I believe this is what you want).  

lib/oxidized/config.rb

	Crash = File.join Root, 'crash'

lib/oxidized/input/ssh.rb

	@log = File.open(Oxidized::Config::Crash + "-#{@node.ip}-ssh", 'w') if CFG.input.debug?

Whats your email? I will send it over.

**Comment by ytti** [comment]
I don't actually mean crash file, I mean the @log file.

My email is 'saku@ytti.fi', thanks a bunch!

**Comment by etern4ever** [comment]
Sent over the log file.  Hopefully that helps figure this out.

**Comment by etern4ever** [comment]
Have you had time to look over the log file?

**Comment by ytti** [comment]
Sorry, forgot this. Just reviewed and was bit disappointed that it wasn't full session log. Now I don't have full confidence that what ever tool was used to edit it, didn't take out special chars.

Anyhow for the the prompt seen, something super simple will work in the model.

This is what we have now.
https://github.com/ytti/oxidized/blob/master/lib/oxidized/model/powerconnect.rb#L3

I think the 'space' part may make it very non-robust.

You could try something like:
prompt /^[\w-]+>$/

It's simplest to match /your/ prompt, but shouldn't match much else. Unless there happens to be line some configuration line without indent which ends with >.

Also perhaps some model contributor has idea what issue you're having, @emjemj @MrRJ45 @skares @sid3windr

**Comment by etern4ever** [comment]
I changed the prompt to that and it still doesn't like it.  Something is odd about these blade switches.

**Comment by ytti** [comment]
It's probably some other issue then. Can you arrange me an remote access to one of the devices? Perhaps lab box attached to a /31 router, so no privileged access anywhere?

This way I could try Oxidized on my own box against it to replicate the issue, and dig deeper.

**Comment by etern4ever** [comment]
Unfortunately, this isn't possible.  These are at a customer site in a production (being blade server switches and all).

I might need to come up with an alternative way of getting these into oxidized.

**Comment by Madpilot0** [comment]
I have the same issue with a PowerConnect switch. However, pretending its a cisco script netted me with something that looked like a config. I have merged the powerconnect and ios files and I'm finally able to see the config files on the web-interface.

**Comment by ytti** [comment]
Ooooh I know what it is. It's same issue we have in CiscoSMB

https://github.com/ytti/oxidized/blob/master/lib/oxidized/model/powerconnect.rb#L31


We define username/password prompt for 'ssh', so we're actually trying to login outside SSH auth process (Some kit actually do that).

Do some PowerConnect versions do that? So ssh auth is 'noauth' and you'll always login, but then get username+password prompt? If not, we can just remove :ssh there, and it'll work.

**Comment by etern4ever** [comment]
yep..  removed :ssh and worked like a charm.

**Comment by ytti** [comment]
Do we know if some PowerConnects authenticate over CLI, even for ssh?

If not, just send pull request and let's take that away.

**Comment by sid3windr** [comment]
My powerconnects (62xx and 53xx) only take direct auth from the SSH client at least.

**Comment by ytti** [comment]
@skares added this, care to comment? Otherwise we rollback.

We can support both SSH methods by some variable if needed, but I'd prefer not to.

**Comment by Madpilot0** [comment]
Removing :ssh from that line worked perfectly for me. 

```
cfg :telnet do
    username /^User( Name)?:/
    password /^\r?Password:/
end
```

Currently it is not giving me a regex error on the prompt part.

**Comment by fnordpojk** [comment]
Was having the same issue on a couple of Dell N4032, removing :ssh fixed it for me too.

**Comment by ytti** [comment]
We have this same issue in powerconnect. Some powerconnects use SSH auth, some no auth and present login prompt.

What is the right fix?

1. variable powerconnect_ssh_cli_auth: true
2. two models? base model and 'xpowerconnect', 'ypowerconnect' which require the base and reopen to define specific behaviour?

**Comment by sid3windr** [comment]
They're likely all "dell powerconnect" but running totally different software? In Observium we for example split "powerconnect radlan" and "powerconnect broadcom".

**Comment by ytti** [comment]
So essentially option 2? But it seems like same model works, except for SSH auth difference.

**Comment by fnordpojk** [comment]
It's even worse with the N4000s, where they behave different depending on which major release of software you're running..

**Comment by broxio** [comment]
experience the same problem with un-predicted special character from CLI  Dell Networking OS , also blank space after colon  , haven't try to find the work around yet.

```
Trying 10.0.x.x...
Connected to 10.0.x.x.
ûLogin: 
```

**Comment by TheBigfoot** [comment]
I also found issues with the prompt, or more with the newlines.

We are running some PowerConnect M6220.
Oxidized Version is 0.14.2 with ruby 2.0.0p598 (2014-11-13) on CentOS 7
```
[root@oxidized]# cat log-192.168.1.1-ssh 

switchHost>
[root@oxidized]# od -x log-192.168.1.1-ssh
0000000 0a0d 7773 7469 6863 6f48 7473 0d3e 000a
0000017
```

The default model results in this prompt error:
```
W, [2016-06-10T15:09:40.137135 #905]  WARN -- : 192.168.1.1 raised Oxidized::PromptUndetect with msg "
switchHost> not matching configured prompt (?-mix:^([\w\s.@-]+[#>]\s?)$)"
```
The linebreak looks wierd to me in front of the hostname.

**Comment by nbarry101** [comment]
We're also having issues with this, and are receiving the following error:

`W, [2016-07-14T12:16:10.034408 #29686]  WARN -- : 192.168.3.248 raised Oxidized::PromptUndetect with msg "
***-**-SWT-02> not matching configured prompt (?-mix:^([\w\s.@-]+[#>]\s?)$)"`

Has anyone had any luck in getting around this?

**Comment by ytti** [comment]
Just for test, could you guys try this version: https://raw.githubusercontent.com/ytti/oxidized/9be5cb4bccad0fcc6fd0b6ab7779cc361fa27620/lib/oxidized/model/procurve.rb

```
cd ~/.config/oxidized
mkdir model
curl -O https://raw.githubusercontent.com/ytti/oxidized/9be5cb4bccad0fcc6fd0b6ab7779cc361fa27620/lib/oxidized/model/procurve.rb
´´´

Should do the trick

**Comment by nbarry101** [comment]
Hi There, I placed the file in the following folder 
/usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/model
and renamed it to 'test.rb' then updated the router.db to 'test', but when running oxidized, get the following:

`$ oxidized -d
I, [2016-07-14T20:45:05.254167 #2357]  INFO -- : Oxidized starting, running as pid 2357
I, [2016-07-14T20:45:05.254754 #2357]  INFO -- : lib/oxidized/nodes.rb: Loading nodes
D, [2016-07-14T20:45:05.254942 #2357] DEBUG -- : resolving DNS for 192.168.2.252...
D, [2016-07-14T20:45:05.293822 #2357] DEBUG -- : lib/oxidized/node.rb: Loading model "test"
D, [2016-07-14T20:45:05.295066 #2357] DEBUG -- : lib/oxidized/model/model.rb Added all to the commands list
D, [2016-07-14T20:45:05.295181 #2357] DEBUG -- : lib/oxidized/model/model.rb Added show version to the commands list
D, [2016-07-14T20:45:05.295274 #2357] DEBUG -- : lib/oxidized/model/model.rb Added show system-information to the commands list
D, [2016-07-14T20:45:05.295360 #2357] DEBUG -- : lib/oxidized/model/model.rb Added show running-config to the commands list
F, [2016-07-14T20:45:05.295799 #2357] FATAL -- : Oxidized crashed, crashfile written in /home/dwmadmin/.config/oxidized/crash
undefined method `new' for nil:NilClass
/usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/manager.rb:18:in `load': undefined method `new' for nil:NilClass (NoMethodError)
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/manager.rb:47:in `add_model'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/node.rb:169:in `resolve_model'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/node.rb:19:in `initialize'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:19:in `new'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:19:in `block (2 levels) in load'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:15:in `each'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:15:in `block in load'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:111:in `synchronize'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:111:in `with_lock'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:10:in `load'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/nodes.rb:106:in `initialize'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/core.rb:14:in `new'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/core.rb:14:in `initialize'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/core.rb:4:in `new'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/core.rb:4:in `new'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/cli.rb:12:in `run'
        from /usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/bin/oxidized:9:in `<top (required)>'
        from /usr/local/rvm/gems/ruby-2.3.0/bin/oxidized:23:in `load'
        from /usr/local/rvm/gems/ruby-2.3.0/bin/oxidized:23:in `<main>'
        from /usr/local/rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `eval'
        from /usr/local/rvm/gems/ruby-2.3.0/bin/ruby_executable_hooks:15:in `<main>'`

Have I missed something?
This is a Dell N2024

**Comment by nbarry101** [comment]
Hi There, 

I've made the change, but received the following:

`
192.168.2.252 raised Oxidized::PromptUndetect with msg "
xx-xx-SWT-02> not matching configured prompt (?m-ix:.*?(\w+# ).*)"
/usr/local/rvm/gems/ruby-2.3.0/gems/oxidized-0.15.0/lib/oxidized/input/telnet.rb:107: warning: constant Net::Telnet::TimeoutError is deprecated
W, [2016-07-14T20:51:26.187384 #2612]  WARN -- : 192.168.2.252 raised Timeout::Error with msg "timed out while waiting for more data"
D, [2016-07-14T20:51:26.187436 #2612] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet failed for 192.168.2.252
D, [2016-07-14T20:51:26.187469 #2612] DEBUG -- : lib/oxidized/job.rb: Config fetched for 192.168.2.252 at 2016-07-14 10:51:26 UTC
W, [2016-07-14T20:51:26.977964 #2612]  WARN -- : 192.168.2.252 status no_connection, retry attempt 1
D, [2016-07-14T20:51:26.978032 #2612] DEBUG -- : lib/oxidized/worker.rb: Jobs 0, Want: 1
D, [2016-07-14T20:51:26.978182 #2612] DEBUG -- : lib/oxidized/worker.rb: Added 192.168.2.252 to the job queue
D, [2016-07-14T20:51:26.978200 #2612] DEBUG -- : lib/oxidized/worker.rb: 1 jobs running in parallel
D, [2016-07-14T20:51:26.978232 #2612] DEBUG -- : lib/oxidized/job.rb: Starting fetching process for 192.168.2.252 at 2016-07-14 10:51:26 UTC
D, [2016-07-14T20:51:26.978318 #2612] DEBUG -- : lib/oxidized/input/ssh.rb: Connecting to 192.168.2.252
D, [2016-07-14T20:51:27.177606 #2612] DEBUG -- : lib/oxidized/input/ssh.rb: expecting /.*?(\w+# ).*/m at 192.168.2.252
`

This is a Dell N2024.
Have I missed something?

Thanks
Nicholas Barry

**Comment by ytti** [comment]
Can you try with this:
https://raw.githubusercontent.com/ytti/oxidized/master/lib/oxidized/model/procurve.rb

You can literally do this:
```
cd ~/.config/oxidized
mkdir model
cd model
curl -O https://raw.githubusercontent.com/ytti/oxidized/master/lib/oxidized/model/procurve.rb
^D
```

Then that version is used, instead of distribution version. Remove it, and distribution version is used.

**Comment by nbarry101** [comment]
No worries - that makes sense!

I've completed this, and the result is the same:

`oxidized
I, [2016-07-14T21:06:46.620027 #2833]  INFO -- : Oxidized starting, running as pid 2833
I, [2016-07-14T21:06:46.620638 #2833]  INFO -- : lib/oxidized/nodes.rb: Loading nodes
I, [2016-07-14T21:06:46.654809 #2833]  INFO -- : lib/oxidized/nodes.rb: Loaded 1 nodes
Puma 2.16.0 starting...
* Min threads: 0, max threads: 16
* Environment: development
* Listening on tcp://127.0.0.1:8888
W, [2016-07-14T21:07:07.115850 #2833]  WARN -- : 192.168.2.252 raised Oxidized::PromptUndetect with msg "
xxxx-xxx-SWT-02> not matching configured prompt (?-mix:^\r?([\w -]+\eE)?([\w.-]+# )$)"
W, [2016-07-14T21:07:07.925177 #2833]  WARN -- : 192.168.2.252 status no_connection, retry attempt 1
`

**Comment by ytti** [comment]
@Ultra2D any thoughts

**Comment by nbarry101** [comment]
Strangely, when telnet is enabled, I get this output:

`!
! Machine Description............... Dell Networking Switch
! System Model ID................... N2024
! Machine Type...................... Dell Networking N2024
! Serial Number..................... xxx
! Manufacturer...................... 0xbc00
! Burned In MAC Address............. F8B1.5669.xxx
! System Object ID.................. 1.3.6.1.4.1.674.10895.3053
! SOC Version....................... BCM56342_A0
! HW Version........................ 5
! CPLD Version...................... 13
!
! unit active      backup      current-active next-active
! ---- ----------- ----------- -------------- --------------
! 1    6.1.2.4     6.1.0.6     6.1.2.4        6.1.2.4
!
! System Description: Dell Networking Switch
! USB Port Power Status:
! ----------------------
! Device Not Present
                      ^
% Invalid input detected at '^' marker.
`

When there is no telnet fallback, no file is created.

Telnet log
`D, [2016-07-14T21:24:15.903062 #3092] DEBUG -- : Telnet: terminal length 0 @192.168.2.252
D, [2016-07-14T21:24:15.970895 #3092] DEBUG -- : lib/oxidized/model/model.rb Collecting commands' outputs
D, [2016-07-14T21:24:15.971013 #3092] DEBUG -- : lib/oxidized/model/model.rb Executing show version
D, [2016-07-14T21:24:15.971121 #3092] DEBUG -- : Telnet: show version @192.168.2.252
D, [2016-07-14T21:24:16.050859 #3092] DEBUG -- : lib/oxidized/model/model.rb Executing show system
D, [2016-07-14T21:24:16.050968 #3092] DEBUG -- : Telnet: show system @192.168.2.252
D, [2016-07-14T21:24:16.153111 #3092] DEBUG -- : lib/oxidized/model/model.rb Executing show running-config
D, [2016-07-14T21:24:16.153220 #3092] DEBUG -- : Telnet: show running-config @192.168.2.252
D, [2016-07-14T21:24:16.222844 #3092] DEBUG -- : lib/oxidized/input/cli.rb Running pre_logout commands at 192.168.2.252
D, [2016-07-14T21:24:16.222941 #3092] DEBUG -- : Telnet: logout @192.168.2.252
D, [2016-07-14T21:24:16.253783 #3092] DEBUG -- : Telnet: exit @192.168.2.252
D, [2016-07-14T21:24:16.253985 #3092] DEBUG -- : lib/oxidized/node.rb: Oxidized::Telnet ran for 192.168.2.252 successfully
D, [2016-07-14T21:24:16.254115 #3092] DEBUG -- : lib/oxidized/job.rb: Config fetched for 192.168.2.252 at 2016-07-14 11:24:16 UTC
I, [2016-07-14T21:24:16.517413 #3092]  INFO -- : Configuration updated for /192.168.2.252
`

(This is using the stock powerconnect.rb file)

**Comment by Ultra2D** [comment]
@ytti not really. I guess it wouldn't hurt to start with a very relaxed regex like the original one for procurve:
```
/.*?(\w+# ).*/m
```
And then change it to
```
/.*?([\w.-]+# ).*/m
```

**Comment by nbarry101** [comment]
Hi @Ultra2D I gave this a whirl, the result is the same:
W, [2016-07-14T22:35:44.717425 #1215]  WARN -- : 192.168.2.252 raised Oxidized::PromptUndetect with msg "
xxxx-xxx-SWT-02> not matching configured prompt (?m-ix:.*?([\w.-]+# ).*)"

**Comment by Ultra2D** [comment]
Sorry, please try ```/.*?([\w.-]+> ).*/m``` (or  ```/.*?([\w.-]+[>#] ).*/m```). I am not sure why the question mark is there though. If that does not work, remove the space.

**Comment by nbarry101** [comment]
I'll give that a whirl, thanks :)

**Comment by nbarry101** [comment]
Hi @Ultra2D sorry but that didn't work either
`
W, [2016-07-15T20:22:49.726156 #27154]  WARN -- : 192.168.2.252 raised Oxidized::PromptUndetect with msg "
BRCC-CSC-SWT-02> not matching configured prompt (?m-ix:.*?([\w.-]+[>#] ).*)"
`

**Comment by Ultra2D** [comment]
Most likely it's just a matter of removing the space (I should have used ```\s?``` in stead of ``` ```. Anyway, I did not look at https://github.com/ytti/oxidized/blob/master/lib/oxidized/model/powerconnect.rb#L3 before. Can you try  ```/.*?([\w\s.@-]+[#>]\s?).*/m``` ? Note that you probably don't want to use this as a final solution. If that does not work, I am at a loss.

**Comment by nbarry101** [comment]
Sorry mate, still not working.
Just in case it helps, here is a screenshot of the output:

![image](https://cloud.githubusercontent.com/assets/5484303/16892940/a177c06c-4b68-11e6-9f32-cd8b01924c01.png)



The blurred out letters are just alpha chars

**Comment by Ultra2D** [comment]
The configured prompt in the screenshot does not match the one in https://github.com/ytti/oxidized/issues/228#issuecomment-232920987 . Did you try ```/.*?([\w\s.@-]+[#>]\s?).*/m```?

**Comment by nbarry101** [comment]
Hi there, yeah I tried that one, but reverted back to the default. I thought the screenshot may help because there is that odd newline in the hostname.

**Comment by Ultra2D** [comment]
Ok, then I am out of ideas.

**Comment by ytti** [comment]
I was able to get config out from @nbarry101 device.

Issues

  * we try to login/password even with SSH. I think some procurves want this, others don't? I removed ':ssh' from the config
  * the enable handling in the model is not eating prompt after inputting enable PW
  * the enable handling does not support empty enable (why have empty enable? Can't you just disable it?).

Anyhow here is the model which appears to work:
[ytti@ytti.fi ~]% cat ~/.config/oxidized/model/powerconnect.rb|pastebinit
http://p.ip.fi/1Qm9

I'm not sure how we should solve all those issues

  * are there many powerconnects, do some of them prompt for username+PW, even on ssh? Should we give some variable to differentiate?
  * should we support empty enable passwords, if so, how? Right now we consider empty value 'nil' (i.e. not defined). How should we be able to communicate in ```router.db```these three things:
    1. enable is on, and has a string
    2. enable is on, and is empty
    3. enable is not on

**Comment by nbarry101** [comment]
That works for me :) thanks so much 
I tried on 5 of out dell switches which worked without issue.

**Comment by ytti** [comment]
enable can reopened on another case but the auth issue was solved by dbab46d5f8228f8dd5

**Comment by jpabl0m** [comment]
Hi. I have AT8000S switches and the ways that describe above doesn't work any of these. Oxidized keeps on telnet connecting and fail because timeout issue. I've changed the prompt but no luck at all. Somebody can resolved that issue or find a way to powerconnect.rb module work with AT8000S models? I still try too many ways but no response at all. Thank you
